### ISSUE TITLE
Correct usage of continuous_eval

### ISSUE THREAD

**Issue by agenius5** [google-research/pegasus#155]
Hi @JingqingZ 

So, continuous_eval basically evaluates on saved checkpoints as the checkpoints becomes available during training itself. I managed to run both the things simultaneously on two colab seesions with gcloud+colab tpu. Now I have two doubts.

First, sometimes checkpoints comes before the previous eval finishes (when I use small batch size training is fast) so how to get both training and eval in proper sync? Right now i run continuous eval after traininig finishes which works fine.

Second and important doubt, which measure do we look at to guide our fine-tunining. Is it perplexity on dev set or the ROUGE scores or both? How did you determine at which step to stop the fine-tuning, evaluate the test set and to get your final numbers?

**Comment by JingqingZ** [comment]
> First, sometimes checkpoints comes before the previous eval finishes (when I use small batch size training is fast) so how to get both training and eval in proper sync? Right now i run continuous eval after traininig finishes which works fine.

This document may explain this better: https://www.tensorflow.org/versions/r1.15/api_docs/python/tf/contrib/training/checkpoints_iterator?hl=en
 
> Second and important doubt, which measure do we look at to guide our fine-tunining. Is it perplexity on dev set or the ROUGE scores or both? How did you determine at which step to stop the fine-tuning, evaluate the test set and to get your final numbers?

How to decide best checkpoint: https://github.com/google-research/pegasus/blob/master/pegasus/bin/evaluate.py#L66

There is no actual "stop" during fine-tuning. We set a maximum fine-tuning step and select the best checkpoint after fine-tuning is done to report scores on test set.

**Comment by agenius5** [comment]
Okay, nice! Got it.

And I suppose all the learning rates, fine-tuning steps etc on various datasets are basically determined by lots of experiments.

**Comment by JingqingZ** [comment]
The hyperparameters of fine-tuning were decided by grid search while others were decided by empirically default commonly used values.
